harness/determined/exec/prep_container.py

```diff
@@ -327,8 +339,8 @@ def do_proxy(sess: api.Session, allocation_id: str) -> None:
     if args.notify_container_running:
         send_container_running_notification(sess, info.allocation_id)
 
-    if args.trial:
-        trial_prep(sess, info)
+    if args.download_context_directory or args.trial:
```

Comment:
If `trial` is deprecated, could you please add a [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning) if it's used?

If `trial` is not deprecated, then does it make sense to not introduce `download_context_directory` and just continue using `trial`?

Reply:
> Does anyone call this script outside of our entrypoints or could it be considered "internal"?

I for sure don't have the context to *actually* know this, but I do think "internal" is a little nuanced. I _believe_ that this is called by (only by?) determined processes from code that, strictly speaking, isn't a part of the `determined-ai/determined` repo. Or at the very least, code that's not a part of `determined-ai/determined/harness`. So in that sense it's at least got a caller not internal to the package it's a part of.

But that's probably a pedantic distinction, and you're asking if we can just remove the flag altogether, no worries about the deprecation dance. I don't even know, here, which team to ask.

Anyway, following the deprecation path, this looks great. Only outstanding suggestion is to remove the `stacklevel` argument, since `prep_container.py` is being called as an executable and it looks funny to assign the warning to Python internals.